server/src/shared/infra/db/entities/auth-code.entity.ts

```diff
@@ -0,0 +1,22 @@
+import { AfterCreate, Entity, Property, Index } from '@mikro-orm/core'
+import { DomainEvents } from '../../../domain/events/domain-events'
+import { UniqueEntityID } from '../../../domain/unique-entity-id'
+import { BaseEntity } from './base.entity'
+
+@Entity()
+export class AuthCodeEntity extends BaseEntity {
+  @Property()
+  clientId!: string
+
+  @Property()
+  @Index()
+  authCode!: string
+
+  // TODO: fix any type
+  @AfterCreate()
+  afterCreate(target: any) {
```

Comment:
instead of any what is this type supposed to be?

Reply:
cc @jeffzh4ng

Reply:
This is a great question. This db hook is used to dispatch a domain event, even though our app doesn't use them as of yet. I looked into MikroORM docs and can't figure out why I typed the param as `any`. Maybe we can create a ticket for this for spring cleaning @xujustinj

Reply:
Haha that's the same conclusion I came to after going through Mikro docs. (Mikro seems to be allergic to code examples)

To properly resolve this we would probably have to go into the source code of Mikro to see what exactly it does with the target

P.S. a better placeholder type is `unknown`

Reply:
can switch it to `unknown` for the timebeing.

Reply:
> This is a great question. This db hook is used to dispatch a domain event, even though our app doesn't use them as of yet. I looked into MikroORM docs and can't figure out why I typed the param as `any`. Maybe we can create a ticket for this for spring cleaning @xujustinj

Now referenced in https://github.com/loolabs/waterpark/issues/156